src/ui/CategoryFacet/CategoryFacetTemplates.ts

```diff
@@ -0,0 +1,39 @@
+import { $$, Dom } from '../../utils/Dom';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { SVGDom } from '../../utils/SVGDom';
+
+export class CategoryFacetTemplates {
+  private listElementTemplate: Dom;
+  private listRoot: Dom;
+
+  constructor() {
+    this.listElementTemplate = this.createListElementTemplate();
+    this.listRoot = $$('ol', { className: 'coveo-facet-values' });
+  }
+
+  public buildListRoot() {
```

Comment:
I think it'd be a nicer API if that class had a single public `build` method that returned one object with everything it build.

Then you could call it with `const {arrow, listElement, listRoot} = categoryFacetTemplate.build()`

Reply:
I agree it would be nicer. But the problem I see is that I never need all of them at once. Also one of the reasons why I made this class was to avoid building a lot of dom elements and use `clone` instead so always building all of them defeats that purpose.

I'm under the impression that it's possible for a lot of values to be built all at once so I was trying to optimise that process. Maybe I'm over optimising here...